app/controllers/solidus_stripe/intents_controller.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+require 'stripe'
+
+class SolidusStripe::IntentsController < Spree::BaseController
+  include Spree::Core::ControllerHelpers::Order
+
+  before_action :load_payment_method
+
+  def payment_confirmation
+    payment = @payment_method.find_in_progress_payment_for(current_order)
+    intent = @payment_method.find_intent_for(payment)
+
+    if params[:payment_intent] != intent.id
+      raise "The payment intent id doesn't match"
+    end
+
+    unless %w[confirm payment].include?(current_order.state.to_s)
+      redirect_to main_app.checkout_state_path(current_order.state)
+      return
+    end
+
+    current_order.state = :payment
+    payment.log_entries.create!(details: { payment_intent: intent.to_json })
+
+    # https://stripe.com/docs/payments/intents?intent=payment
+    case intent.status
+    when 'requires_payment_method'
+      ensure_state_is(current_order, :payment)
+      ensure_state_is(payment, :processing)
+    when 'requires_confirmation', 'requires_action', 'processing'
+      ensure_state_is(payment, :processing)
+      current_order.next!
+    when 'requires_capture'
+      payment.pend! unless payment.pending?
+      current_order.next!
+      ensure_state_is(current_order, :confirm)
+      ensure_state_is(payment, :pending)
+    when 'succeeded'
+      payment.completed! unless payment.completed?
```

Comment:
@elia, after local testing, I've found that this case is not covered by tests or manual checkout. Am I missing something?

Reply:
It's missing testing with autocapture on